linera-faucet/server/src/lib.rs

```diff
@@ -113,58 +116,126 @@ struct BatchProcessor<C: ClientContext> {
     config: BatchProcessorConfig,
     context: Arc<Mutex<C>>,
     client: ChainClient<C::Environment>,
-    faucet_storage: Arc<Mutex<FaucetStorage>>,
+    faucet_storage: Arc<FaucetStorage>,
     pending_requests: Arc<Mutex<VecDeque<PendingRequest>>>,
     request_notifier: Arc<Notify>,
 }
 
 /// Persistent mapping of account owners to chain descriptions
-#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+#[derive(Debug, Clone)]
 struct FaucetStorage {
     /// Maps account owners to their corresponding chain descriptions
-    owner_to_chain: HashMap<AccountOwner, ChainDescription>,
+    owner_to_chain: Arc<papaya::HashMap<AccountOwner, ChainDescription>>,
+    sender: Arc<watch::Sender<()>>,
+    _save_join_handle: Arc<Mutex<Option<JoinHandle<()>>>>,
```

Comment:
Why is the underscore needed? 🤔
```suggestion
    save_join_handle: Arc<Mutex<Option<JoinHandle<()>>>>,
```